### PR TITLE
Support reference tokens

### PR DESCRIPTION
**Pull request by mondo192**
This change adds the missing field `include_reference_token` to the jfrog-go-client. This allows users to create aliases for access tokens as this was not possible with the client. Only the REST API.

Please see the associated open issue [here](https://github.com/jfrog/jfrog-client-go/issues/727) for more details.

- [x] All [tests](https://github.com/jfrog/jfrog-client-go#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [ ] All [static analysis checks](https://github.com/jfrog/jfrog-client-go/actions/workflows/analysis.yml) passed.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----

### CODE REVIEW

File: tests/accesstokens_test.go [hunk 1]
```diff
@@ -40,6 +41,21 @@ func testRefreshTokenTest(t *testing.T) {
 	assert.Equal(t, token.ExpiresIn, newToken.ExpiresIn, "New access token's expiration is different from original one")
 }
 
+func testReferenceTokenTest(t *testing.T) {
+	// Create token
+	tokenParams := createRefreshableAccessTokenParams(testExpiredInSeconds)
+	token, err := testsAccessTokensService.CreateAccessToken(tokenParams)
+	assert.NoError(t, err)
+	// Refernece token
```
Comment (yahavi) [inline tests/accesstokens_test.go:49]:
```suggestion
	// Reference token
```

File: tests/accesstokens_test.go [hunk 2]
```diff
@@ -40,6 +41,21 @@ func testRefreshTokenTest(t *testing.T) {
...
+	referenceTokenParams := createReferenceAccessTokenParams(token)
+	newToken, err := testsAccessTokensService.RefreshAccessToken(referenceTokenParams.CommonTokenParams)
+	assert.NoError(t, err)
+	// Validate
+	assert.NotEqual(t, token.AccessToken, newToken.AccessToken, "New access token is identical to original one")
+	assert.NotEqual(t, token.IsReferenceToken, newToken.IsReferenceToken, "New reference token is identical to original one")
```
Comment (yahavi) [inline tests/accesstokens_test.go:55]: This fails the test. I'm not sure that the "include_reference_token" is included in the response from Xray.

https://github.com/jfrog/jfrog-client-go/actions/runs/4609825075/jobs/8147514091?pr=729#step:6:61

**Review by yahavi — CHANGES_REQUESTED** [review]
Thanks for this contribution @mondo192!
Please see my inline comments.

File: auth/authutils.go [hunk 3]
```diff
@@ -16,14 +16,15 @@ type CreateTokenResponseData struct {
...
-	Audience     string `json:"audience,omitempty"`
+	Scope            string `json:"scope,omitempty"`
+	AccessToken      string `json:"access_token,omitempty"`
+	ExpiresIn        int    `json:"expires_in,omitempty"`
+	TokenType        string `json:"token_type,omitempty"`
+	Refreshable      *bool  `json:"refreshable,omitempty"`
+	RefreshToken     string `json:"refresh_token,omitempty"`
+	GrantType        string `json:"grant_type,omitempty"`
+	Audience         string `json:"audience,omitempty"`
+	IsReferenceToken *bool  `json:"include_reference_token,omitempty"`
```
Comment (yahavi) [inline auth/authutils.go:27]:
```suggestion
	IncludeReferenceToken *bool  `json:"include_reference_token,omitempty"`
```

**Review by yahavi — CHANGES_REQUESTED** [review]
Just to be compatible with API. Change this and we'll merge.

**Review by yahavi — APPROVED** [review]

**Comment by yahavi** [comment]
Hi @mondo192,
After more thought, I made some changes to this PR - 
* Add `include_reference_token` **only** to CreateTokenParams
* Add the generated `reference_token` (ReferenceToken) to the response
* Allow RefreshAccessToken() to receive reference the include_reference_token flag
* Add test
* Document in the README

Please LMK what you think.

**Review by eyalbe4 — APPROVED** [review]

**Pull request merged**